pyvista/plotting/lights.py

```diff
@@ -0,0 +1,466 @@
+                       'scenelight': LightType.SCENE_LIGHT,
+                      }
+            try:
+                light_type = mapping[type_normalized]
+            except KeyError:
+                raise ValueError(f'Invalid ``light_type`` "{light_type_orig}"') from None
+        elif not isinstance(light_type, int):
+            raise TypeError('Parameter ``light_type`` must be int or str,'
+                            f' not {type(light_type)}.')
+        # LightType is an int subclass
+
+        self.light_type = light_type
+
+        # TODO: more init kwargs (focal_point, cone angle etc.)
+        # TODO: default Positional state?
+        # TODO: should setting attenuation and cone angle automatically switch to positional?
+        # TODO: should color and point and direction_angle have more flexible signatures? (only for non-properties)
+        # TODO: ndarray type and shape and size checking for color and point
+
+    def __repr__(self):
+        """Print a repr specifying the id of the light and its light type."""
+        return (f'<{self.__class__.__name__} ({self.light_type}) at {hex(id(self))}>')
+
+    #### Properties ####
+
+    @property
+    def ambient_color(self):
+        """Return the ambient color of the light."""
+        return self.GetAmbientColor()
+
+    @ambient_color.setter
+    def ambient_color(self, color):
+        """Set the ambient color of the light."""
+        self.SetAmbientColor(color)
+
+    @property
+    def diffuse_color(self):
+        """Return the diffuse color of the light."""
+        return self.GetDiffuseColor()
+
+    @diffuse_color.setter
+    def diffuse_color(self, color):
+        """Set the diffuse color of the light."""
+        self.SetDiffuseColor(color)
+
+    @property
+    def specular_color(self):
+        """Return the specular color of the light."""
+        return self.GetSpecularColor()
+
+    @specular_color.setter
+    def specular_color(self, color):
+        """Set the specular color of the light."""
+        self.SetSpecularColor(color)
+
+    # TODO: implement light.color = ... using setattr? no guarantee that a getter would always make sense, so property won't work!
+
+    @property
+    def position(self):
+        """Return the position of the light.
+
+        Note: the position is defined in the coordinate space indicated
+        by the light's transformation matrix (if it exists). To get the
+        light's world space position, use the ``world_position`` property.
+
+        """
+        return self.GetPosition()
+
+    @position.setter
+    def position(self, pos):
+        self.SetPosition(pos)
+
+    @property
+    def world_position(self):
+        # TODO: is this name and configuration OK? Same for world_focal_point
+        # TODO: can a transformation matrix happen accidentally? If not, perhaps we can just not expose these at all!
+        """Return the world space position of the light."""
+        return self.GetTransformedPosition()
```

Comment:
```suggestion
    @property
    def world_position(self):
        """Return the world space position of the light."""
        return self.GetTransformedPosition()
```

I'm happy with the name, though we should detail that we're returning the value from `GetTransformedPosition` from VTK.  This needs an example as well.